Scripts/Prerequisite.ps1

```diff
@@ -0,0 +1,96 @@
+param(
+    [Parameter(Mandatory = $true)]
+    [string]$subscriptionId,
+    [Parameter(Mandatory = $true)]
+    [string]$resourceGroupName,
+    [Parameter(Mandatory = $true)]
+    [string]$keyVaultName,
+    [Parameter(Mandatory = $true)]
+    [string]$location,
+    [Parameter(Mandatory = $true)]
+    [string]$certName,
+    [Parameter(Mandatory = $true)]
+    [string]$keySecretForServicePrincipal,
+    [Parameter(Mandatory = $true)]
+    [string]$servicePrincipalName,
+    [Parameter(Mandatory = $true)]
+    [string]$tenantId,
+    [Parameter(Mandatory = $true)]
+    [string]$serviceConnectionName,
+    [Parameter(Mandatory = $true)]
+    [string]$organizationName,
+    [Parameter(Mandatory = $true)]
+    [string]$projectName
+)
+
+try {
+
+  $isLoggedIn = az account show 
+  if ($null -eq $isLoggedIn) {
+      Write-Host "logging In..." -ForegroundColor White
+      az login --use-device-code
+  }
+  
+  Write-Host "Initiating pre-requisites validation..." -ForegroundColor White
+  $subscriptionName = az account list --query "[?id=='$subscriptionId'].name" -o tsv
+  if ($null -eq $subscriptionName) {
+      Write-Warning "You don't have access to subcription: $subscriptionId ."
+      exit
+  }
+  else {
+      Write-Host "Switching to subscription: $subscriptionId" -ForegroundColor White
+      az account set -s $subscriptionId 
+  }
+    
+  $rgAccess = az group show --name $resourceGroupName --query "name" -o tsv
+    
+  if ($null -eq $rgAccess) {
+      Write-Warning "You don't have access to required resource group: $resourceGroupName. please elevate your roles via PIM and try again."
```

Comment:
update the message as not every team in Microsoft is using PIM elevations, keep it generic